File: maven-html-jdbc-cab-booking/src/main/java/com/vaishnavi/cab/booking/model/Rating.java
package com.vaishnavi.cab.booking.model;

import lombok.*;

@Getter
@Setter
@ToString
@AllArgsConstructor
@Data

public class Rating {private int ratingId;
    private int rideId;
    private int userId;
    private int driverId;
    private int rating;
    private String review;

    public Rating(int ratingId, int rideId, int userId, int driverId, int rating, String feedback) {
    }
}

File: src/main/webapp/WEB-INF/rating.js
function rateDriver() {
    const uid = document.getElementById("ruserId").value;
    const did = document.getElementById("rdriverId").value;
    const rid = document.getElementById("rrideId").value;
    const rating = document.getElementById("rating").value;
    const review = document.getElementById("review").value;

    fetch('RatingController', {
        method: 'POST',
        headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
        body: `action=rate&userId=${uid}&driverId=${did}&rideId=${rid}&rating=${rating}&review=${review}`
    }).then(res => res.text())
      .then(() => {
        alert("Thank you for your feedback!");
        getAllRatings();
    });
}

function getAllRatings() {
    fetch('RatingController?action=getAll')
        .then(res => res.json())
        .then(ratings => {
            let list = "<h3>All Ratings</h3><ul>";
            ratings.forEach(r => list += `<li>${r.ratingId} - ${r.rating}⭐ (${r.review})</li>`);
            document.getElementById("ratingList").innerHTML = list + "</ul>";
        });
}

window.onload = getAllRatings;
